_posts/2023-07-03-introducing-palletjack.md

```diff
@@ -0,0 +1,80 @@
+
+## Step Two: Transform the Data
+{: .text-left}
+
+Unfortunately, external data is rarely plug and play. We often have to massage it into form, like joining the rental assistance data to the zip code geometries or adding the proper county to the recycling data.
+
+Every project is going to have it's own specific data transformations—some simple, some complex. This is where you should be spending all your coding time and energy. You'll take the DataFrame produced by the extract step and modify it to your heart's content.
+
+While palletjack can't magically figure out your transformations for you, it does have some helper methods in the [`palletjack.transform`](https://agrc.github.io/palletjack/palletjack/transform.html) module to simplify some already-solved problems, like geocoding addresses with the [UGRC Web API Geocoder](https://gis.utah.gov/data/address-geocoders-locators/#WebAPI). It also includes some methods for cleaning up your DataFrame after you've finished your project-specific transformations, preparing your data for loading into an ArcGIS Online Hosted Feature Service.
+
+## Step Three: Load the Data into an Existing Hosted Feature Service
+{: .text-left}
+
+Now that our new DataFrame looks the way we want it, it's time to update our live data. The [`palletjack.load.FeatureServiceUpdater`](https://agrc.github.io/palletjack/palletjack/load.html#palletjack.load.FeatureServiceUpdater) class provides four methods for each possible updating scenario: adding new features, updating existing features, removing features we don't need any longer, and finally just truncating the entire service and loading completely new data. It's your job to figure out which one (or more) methods to use. Each one takes a DataFrame as its input,
+
+Before palletjack, I had to reinvent the wheel every time we needed to update a feature service. I'd often go copy/paste code from a different project—and then spend hours troubleshooting when it wouldn't work with this particular dataset. Later, when I'd update the uploading code in the first project, I'd never remember to update the second.
+
+palletjack fixes this by doing all the heavy lifting for you. You just have to call the right methods. We've spent lots of development time identifying and fixing odd corner cases—datetimes, weird data types, handling null values, and a lot more—so that you can spend less time troubleshooting and more time solving the problems that really matter. If there is something wrong with your data that will prevent it from uploading properly, the palletjack methods will return a helpful error message with hints on how to fix it. Unless it's a problem we've not seen before. Then ~you're on your own~ please log an issue so we can fix it!
```

Comment:
**polish**: correct strike through and link to the new issue
```suggestion
palletjack fixes this by doing all the heavy lifting for you. You just have to call the right methods. We've spent lots of development time identifying and fixing odd corner cases—datetimes, weird data types, handling null values, and a lot more—so that you can spend less time troubleshooting and more time solving the problems that really matter. If there is something wrong with your data that will prevent it from uploading properly, the palletjack methods will return a helpful error message with hints on how to fix it. Unless it's a problem we've not seen before. Then ~~you're on your own~~ please [log an issue](https://github.com/agrc/palletjack/issues/new) so we can fix it!
```